Covscript/test/precedence.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { expectEx } from "./common";
+
+describe("Arithmetic Operator precedence Tests", () => {
+  describe("Left to right - same precedence", () => {
+    expectEx("1 / 3 * 2", { "*": [{ "/": [1, 3] }, 2] });
+    expectEx("1 * 3 / 2", { "/": [{ "*": [1, 3] }, 2] });
+    expectEx("1 + 3 - 2", { "-": [{ "+": [1, 3] }, 2] });
+    expectEx("1 - 3 + 2", { "+": [{ "-": [1, 3] }, 2] });
+  });
+
+  describe("Different precedence", () => {
+    expectEx("1 * 3 + 1", { "+": [{ "*": [1, 3] }, 1] });
+    expectEx("1 + 3 * 1", { "+": [1, { "*": [3, 1] }] });
+  });
+});
+
+describe("Logic Operator precedence Tests", () => {
+  expectEx("1 AND 2 OR 3 AND 4", { or: [{ and: [1, 2] }, { and: [3, 4] }] });
+  expectEx("1 OR 2 AND 3 OR 4", { or: [{ or: [1, { and: [2, 3] }] }, 4] });
+  expectEx("!1 AND !2 AND !3 OR 4", {
+    or: [{ and: [{ and: [{ "!": 1 }, { "!": 2 }] }, { "!": 3 }] }, 4],
+  });
+});
+
+describe("Paranteses Tests", () => {
+  expectEx("1 * (3 + 1)", { "*": [1, { "+": [3, 1] }] });
+  expectEx("1 OR (3 AND 1)", { or: [1, { and: [3, 1] }] });
```

Comment:
Is it intentional that the parentheses are around the "AND", which has precedence anyway? Maybe add for both cases also the other parentheses pair, just to be sure that it always works. "(1*3)+1" and "(1 OR 3) AND 1"

Reply:
Done.